jina/orchestrate/deployments/__init__.py

```diff
@@ -691,6 +710,36 @@ def _set_pod_args(args: Namespace) -> Dict[int, List[Namespace]]:
                     # if there are no shards/replicas, we dont need to distribute ports randomly
                     # we should rather use the pre assigned one
                     args.port = args.port
+                    _args.port_monitoring = args.all_port_monitoring[0]
+                elif args.shards == 1 and args.replicas > 1:
+                    _args.port_monitoring = (
```

Comment:
can't this be generalized as having one counter with a max_num_ports? and some custom iteration?

Reply:
Yes probably I will do some refactoring